File: src/main/java/hello/UserController.java
package hello;

import java.util.List;
import java.util.ArrayList;

import hello.Repository.UserRepository;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.bind.annotation.RequestMethod;

@RestController
public class UserController {

    List<User> users = new ArrayList<User>();

    public UserController() {
        UserRepository userRepository = new UserRepository();
        this.users = userRepository.getUsers();
    }

    @RequestMapping("/user")
    public List<User> getAllUser() {
        return users;
    }

    @RequestMapping("/user/{id}")
    public User getSingleUser(@PathVariable(value="id") int id) {
        User result = null;
        for(User currentUser: this.users) {
            if (currentUser.getId() == id) {
                result = currentUser;
                break;
            }
        }
        return result;
    }

    @RequestMapping(value = "/user/{id}", method = RequestMethod.DELETE)
    public void removeSingleUser(@PathVariable(value="id") int id) {
        for(User currentUser: this.users) {
            if (currentUser.getId() == id) {
                this.users.remove(currentUser);
                break;
            }
        }
    }

    @RequestMapping("/user/search")
    public List<User> searchUser(@RequestParam(value = "query") String query) {
        List<User> result = new ArrayList<User>();
        for(User currentUser: this.users) {
            if (currentUser.getName().contains(query)) {
                result.add(currentUser);
            }
        }
        return result;
    }

}

File: src/main/java/hello/UserController.kt
package hello

import hello.Builder.UserBuilder
import hello.Repository.UserRepository
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import org.springframework.web.server.ResponseStatusException
import java.util.*

@RestController
class UserController {
    var users: MutableList<User> = ArrayList()
    @get:RequestMapping("/user")
    val allUser: List<User>
        get() = users

    @RequestMapping(value = ["/user"], method = [RequestMethod.POST])
    fun addSingleUser(@RequestParam(value = "name") name: String?, @RequestParam(value = "age") age: Int) {
        val userCount = users.size
        val user = UserBuilder().id(userCount + 1).name(name).age(age).build()
        users.add(user)
    }

    @RequestMapping("/user/{id}")
    fun getSingleUser(@PathVariable(value = "id") id: Int): User {
        var result: User? = null
        for (currentUser in users) {
            if (currentUser.id == id) {
                result = currentUser
                break
            }
        }
        if (result == null) {
            throw ResponseStatusException(HttpStatus.NOT_FOUND, "User Not Found")
        }
        return result
    }

    @RequestMapping(value = ["/user/{id}"], method = [RequestMethod.DELETE])
    fun removeSingleUser(@PathVariable(value = "id") id: Int) {
        for (currentUser in users) {
            if (currentUser.id == id) {
                users.remove(currentUser)
                break
            }
        }
    }

    @RequestMapping("/user/search")
    fun searchUser(@RequestParam(value = "query") query: String): List<User> {
        if (query.isBlank() || query.isEmpty()) {
            throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Search Query Can Not be Empty!")
        }
        val result: MutableList<User> = ArrayList()
        for (currentUser in users) {
            if (currentUser.name!!.contains(query)) {
                result.add(currentUser)
            }
        }
        return result
    }

    init {
        val userRepository = UserRepository()
        users = userRepository.users
    }
}